app/legacy_lib/query_payments.rb

```diff
@@ -168,20 +168,22 @@ def self.select_to_filter_search(npo_id, query)
     if query[:search].present?
       expr = SearchVector.query(query[:search], expr)
     end
-    if ['asc', 'desc'].include? query[:sort_amount]
-      expr = expr.order_by("payments.gross_amount #{query[:sort_amount]}")
-    end
-    if ['asc', 'desc'].include? query[:sort_date]
-      expr = expr.order_by("payments.date #{query[:sort_date]}")
-    end
-    if ['asc', 'desc'].include? query[:sort_name]
-      expr = expr.order_by("coalesce(NULLIF(supporters.name, ''), NULLIF(supporters.email, '')) #{query[:sort_name]}")
-    end
-    if ['asc', 'desc'].include? query[:sort_type]
-      expr = expr.order_by("payments.kind #{query[:sort_type]}")
-    end
-    if ['asc', 'desc'].include? query[:sort_towards]
-      expr = expr.order_by("NULLIF(payments.towards, '') #{query[:sort_towards]}")
+    unless (query[:campaign_id].present? || query[:event_id].present?) # if we need to add the reverse query, we can't add this here.
```

Comment:
Responding to your comment on this line: If we can't add the reverse query here, where would we add it? Or would we need to add another conditional to this file?

Reply:
The problem comes from the fact that the reverse query requires doing a union. We actually do the same sort in two places in the query (we found it was more performant). The problem is that when you do a union, the ordering clauses have to be structured different. While it is probably best to work on updating the query so we can do the sort in both places, we need to get it working again first. This works around the problem by skipping the inner sort when we add a union the query. The outer sort still happens which means the results continue to be correct albeit a bit slower than we'd like.

Reply:
I see!  Ok, this makes sense to me. Thank you